datafusion/optimizer/src/optimize_projections/mod.rs

```diff
@@ -472,11 +471,8 @@ fn merge_consecutive_projections(proj: Projection) -> Result<Transformed<Project
 
     // Count usages (referrals) of each projection expression in its input fields:
     let mut column_referral_map = HashMap::<&Column, usize>::new();
-    for columns in expr.iter().map(|expr| expr.column_refs()) {
-        for col in columns.into_iter() {
-            *column_referral_map.entry(col).or_default() += 1;
-        }
-    }
+    expr.iter()
```

Comment:
I am confirming my understanding of this change in behavior. 

The old code counts would treat a projection with two expression like `[a + a, b]` as having only a single reference to `a`. After the change, it would correctly identify 2 occurrences of `a.

Reply:
Exactly.